test/libevmasm/EVMAssemblyTest.cpp

```diff
@@ -0,0 +1,181 @@
+
+#include <test/Common.h>
+
+#include <libevmasm/Disassemble.h>
+#include <libevmasm/EVMAssemblyStack.h>
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/trim.hpp>
+
+#include <range/v3/view/map.hpp>
+
+using namespace std::string_literals;
+using namespace solidity;
+using namespace solidity::test;
+using namespace solidity::evmasm;
+using namespace solidity::evmasm::test;
+using namespace solidity::frontend;
+using namespace solidity::frontend::test;
+using namespace solidity::langutil;
+using namespace solidity::util;
+
+std::vector<std::string> const EVMAssemblyTest::c_outputLabels = {
+	"InputAssemblyJSON",
+	"Assembly",
+	"Bytecode",
+	"Opcodes",
+	"SourceMappings",
+};
+
+std::unique_ptr<TestCase> EVMAssemblyTest::create(Config const& _config)
+{
+	return std::make_unique<EVMAssemblyTest>(_config.filename);
+}
+
+EVMAssemblyTest::EVMAssemblyTest(std::string const& _filename):
+	EVMVersionRestrictedTestCase(_filename)
+{
+	m_source = m_reader.source();
+	m_expectation = m_reader.simpleExpectations();
+
+	if (boost::algorithm::ends_with(_filename, ".asmjson"))
+		m_assemblyFormat = AssemblyFormat::JSON;
+	else if (boost::algorithm::ends_with(_filename, ".asm"))
+		m_assemblyFormat = AssemblyFormat::Plain;
+	else
+		BOOST_THROW_EXCEPTION(std::runtime_error("Not an assembly test: \"" + _filename + "\". Allowed extensions: .asm, .asmjson."));
+
+	m_selectedOutputs = m_reader.stringSetting("outputs", "Assembly,Bytecode,Opcodes,SourceMappings");
+	OptimisationPreset optimizationPreset = m_reader.enumSetting<OptimisationPreset>(
+		"optimizationPreset",
+		{
+			{"none", OptimisationPreset::None},
+			{"minimal", OptimisationPreset::Minimal},
+			{"standard", OptimisationPreset::Standard},
+			{"full", OptimisationPreset::Full},
+		},
+		"none"
+	);
+	m_optimizerSettings = Assembly::OptimiserSettings::translateSettings(OptimiserSettings::preset(optimizationPreset));
+	m_optimizerSettings.expectedExecutionsPerDeployment = m_reader.sizetSetting(
+		"optimizer.expectedExecutionsPerDeployment",
+		m_optimizerSettings.expectedExecutionsPerDeployment
+	);
+
+	auto const optimizerComponentSetting = [&](std::string const& _component, bool& _setting) {
+		_setting = m_reader.boolSetting("optimizer." + _component, _setting);
+	};
+	optimizerComponentSetting("inliner", m_optimizerSettings.runInliner);
+	optimizerComponentSetting("jumpdestRemover", m_optimizerSettings.runJumpdestRemover);
+	optimizerComponentSetting("peephole", m_optimizerSettings.runPeephole);
+	optimizerComponentSetting("deduplicate", m_optimizerSettings.runDeduplicate);
+	optimizerComponentSetting("cse", m_optimizerSettings.runCSE);
+	optimizerComponentSetting("constantOptimizer", m_optimizerSettings.runConstantOptimiser);
+
+	// TODO: Enable when assembly import for EOF is implemented.
+	if (CommonOptions::get().eofVersion().has_value())
+		m_shouldRun = false;
```

Comment:
The test case can easily be extended support EOF, but we need to implement assembly import for EOF first.